scipy/signal/tests/test_signaltools.py

```diff
@@ -194,22 +231,25 @@ def test_convolve_method(self, n=100):
         # These are random arrays, which means test is much stronger than
         # convolving testing by convolving two np.ones arrays
         np.random.seed(42)
-        array_types = {'i': np.random.choice([0, 1], size=n),
-                       'f': np.random.randn(n)}
+        array_types = {'i': xp.asarray(np.random.choice([0, 1], size=n)),
+                       'f': xp.asarray(np.random.randn(n))}
         array_types['b'] = array_types['u'] = array_types['i']
         array_types['c'] = array_types['f'] + 0.5j*array_types['f']
 
         for t1, t2, mode in args:
-            x1 = array_types[np.dtype(t1).kind].astype(t1)
-            x2 = array_types[np.dtype(t2).kind].astype(t2)
+            if is_cupy(xp) and t2 == 'float16':
```

Comment:
We shouldn't be testing non-standard dtypes with other array types, so no `float16` support (we decided this in another discussion already).

Reply:
Do you happen to have link to that other discussion? pytorch does not have uint{16, 32,64}, I'd rather sync with then other decision here.

Reply:
not really, that's O(1,000) comments on a set of PRs to trawl through